syntax = "proto3";

service PublishTopic {
  rpc publishRequest (topicData) returns (Acknowledge) {}

  rpc subscribeRequest (topicSubscribe) returns (Acknowledge) {}

  rpc giveIps (topic) returns (stream ips) {}

  rpc publish (topicData) returns (Acknowledge) {}

  rpc getFrontIp (empty) returns (ips) {}

  rpc registerIp (ips) returns (Acknowledge) {}

}


message topicData {
  string topic = 1;
  string data = 2;
}

message topicSubscribe {
  string topic = 1;
  string client_ip = 2;
}

message topic {
  string topic = 1;
}

message ips {
  string ip = 1; 
}

message Acknowledge {
  string ack = 1;
}

message empty {
}